astroquery/noao/core.py

```diff
@@ -0,0 +1,78 @@
+"""
+Provide astroquery API access to OIR Lab Astro Data Archive (natica).
+
+This does DB access through web-services. It allows query against ALL
+fields in FITS file headers.
+
+Also possible but not provided here: query through Elasticsearch DSL.  ES will
+be much faster but much more limitted in what can be used in query.  """
+
+# Python library
+# External packages
+import requests
+import astropy.table
+# Local packages
+
+# 3. local imports - use relative imports
+# commonly required local imports shown below as example
+# all Query classes should inherit from BaseQuery.
+from ..query import BaseQuery
+# has common functions required by most modules
+from ..utils import commons
+# prepend_docstr is a way to copy docstrings between methods
+from ..utils import prepend_docstr_nosections
+# async_to_sync generates the relevant query tools from _async methods
+from ..utils import async_to_sync
+from ..utils.class_or_instance import class_or_instance
+# import configurable items declared in __init__.py
+from . import conf
+
+
+__all__ = ['Noao', 'NoaoClass']  # specifies what to import
+
+
+@async_to_sync
+class NoaoClass(BaseQuery):
+
+    NAT_URL = conf.server
+    ADS_URL = f'{NAT_URL}/api/adv_search/fasearch'
+    SIA_URL = f'{NAT_URL}/api/sia/voimg'
+
+    def __init__(self, which='voimg'):
+        """ set some parameters """
+        # Change following to match current https://<root>/api/version
+        KNOWN_GOOD_API_VERSION = 2.0
+        # #!response = self._request('GET', f'{self.NAT_URL}/api/version',
+        # #!                         cache=False)
+        response = requests.get(f'{self.NAT_URL}/api/version')
```

Comment:
This is what's triggering your `remote_data` access.  You can't do a remote call in the `__init__` step, because `__init__` happens at import time.  We actually forbid that explicitly.

A good alternative might be to do something like add a `.api_version` property that performs a (cached) query to NOAO to determine the API version the first time it's called.

Reply:
Thanks the explanation.  I was unlikely to ever find that!  And I love the alternative. I'll implement.